js/source/worker_tile.js

```diff
@@ -106,23 +106,15 @@ class WorkerTile {
                 this.redoPlacementAfterDone = false;
             }
 
-            const featureIndex_ = featureIndex.serialize();
-            const collisionTile_ = collisionTile.serialize();
-            const collisionBoxArray = this.collisionBoxArray.serialize();
-            const symbolInstancesArray = this.symbolInstancesArray.serialize();
-            const symbolQuadsArray = this.symbolQuadsArray.serialize();
-            const nonEmptyBuckets = util.values(buckets).filter(isBucketNonEmpty);
-
+            const transferables = [];
             callback(null, {
-                buckets: nonEmptyBuckets.map(serializeBucket),
-                featureIndex: featureIndex_.data,
-                collisionTile: collisionTile_.data,
-                collisionBoxArray: collisionBoxArray,
-                symbolInstancesArray: symbolInstancesArray,
-                symbolQuadsArray: symbolQuadsArray
-            }, getTransferables(nonEmptyBuckets)
-                .concat(featureIndex_.transferables)
-                .concat(collisionTile_.transferables));
+                buckets: serializeBuckets(util.values(buckets), transferables),
+                featureIndex: featureIndex.serialize(transferables),
+                collisionTile: collisionTile.serialize(transferables),
+                collisionBoxArray: this.collisionBoxArray.serialize(),
+                symbolInstancesArray: this.symbolInstancesArray.serialize(),
+                symbolQuadsArray: this.symbolQuadsArray.serialize()
```

Comment:
Quick question, why are the latter three not put into transferables? Are they already there indirectly through the symbol buckets?